test/Guardian.t.sol

```diff
@@ -0,0 +1,367 @@
+/**
+ * Created by Arcadia Finance
+ * https://www.arcadia.finance
+ *
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity ^0.8.13;
+
+import "../lib/forge-std/src/Test.sol";
+import {Guardian} from "../src/security/Guardian.sol";
+
+contract LendingPoolMockup is Guardian {
+    uint256 public totalSupply;
+    uint256 public totalBorrow;
+
+    //    constructor()
+    //    {}
+
+    function depositGuarded(uint256 supply) external whenDepositNotPaused {
+        totalSupply += supply;
+    }
+
+    function borrowUnguarded(uint256 borrow) external {
+        totalBorrow += borrow;
+    }
+
+    function borrowGuarded(uint256 borrow) external whenBorrowNotPaused {
+        totalBorrow += borrow;
+    }
+
+    function reset() external onlyOwner {
+        totalSupply = 0;
+        totalBorrow = 0;
+    }
+}
+
+contract GuardianUnitTest is Test {
+    using stdStorage for StdStorage;
+
+    LendingPoolMockup lendingPool;
+    address newGuardian = address(1);
+    address guardian = address(2);
+    address owner = address(3);
+    address nonOwner = address(4);
+    address user = address(5);
+
+    constructor() {
+        vm.startPrank(owner);
+        lendingPool = new LendingPoolMockup();
+        lendingPool.changeGuardian(guardian);
+        vm.stopPrank();
+    }
+
+    function setUp() public virtual {
+        // Reset the lending pool variables
+        vm.startPrank(owner);
+        lendingPool.reset();
+        vm.stopPrank();
+        // Reset: the lending pool pauseTimestamp
+        stdstore.target(address(lendingPool)).sig(lendingPool.pauseTimestamp.selector).checked_write(uint256(0));
```

Comment:
should be set at 0 upon contract creation

Reply:
It is here because if the one test alters the pause state and timestamp. It shouldn't pass to the following test. Instead of redeploying the contract at each setup this will increase the test speed to just reset the parts that are relevant.